### DISCUSSION TITLE
Where are the logfiles saved?

### DISCUSSION THREAD

**Discussion by thrhymes** [blakeblackshear/frigate#10800] [General]
Hello! 

Frigate is an amazing tool, but i run intpo problems with missing recordings and since the cameras where available and the stream too i would like to read the logfiles or - better - get the logs sent to a logserver, so i can run ai tools over them to correlate it with 

i dont find any information in the documentation.... 
Where are the logs? 
Is there a way to forward them to a network-log-server?

**Comment by NickM-27** [comment]
Log files are retrievable via api and the webui, they are stored internally in /dev/shm/logs but the api would definitely be the better place to get it

**Reply by thrhymes** [reply]
Oh, thanks for the answer. 
So i could fetch the logs frequently and feed them into a log-server... ?

There is no GET /api/log in the HTTP api Documentation... how would i use the api to retreive the logs?

i would rather put the logservers data into the config to be more responsive but i guess i can make due with the api too...

**Reply by NickM-27** [reply]
I don't think we'd allow configuring a separate log server, we use the logs internally for a number of things. 

Looks like it's not on the docs, you can visit the webUI and look at the logs and see what the api calls are via desktop dev tools

**Reply by thrhymes** [reply]
ok, thanks
\_(O.o)_/

i guess i have to write my own log.py

**Comment by blakeblackshear** [comment]
I would recommend using a standard log driver designed to collect logs for docker containers. Frigate outputs to stdout just like any normal container. Any off the shelf solutions to collect container logs will work fine.

**Reply by thrhymes** [reply]
oh, ok, that makes sense,.. was too focused on frigate itself.... 

 will try that then...

**Reply by ajbastien** [reply]
API: GET /api/logs/frigate?start=[n]
start is the starting log line.  WebUI uses this to get all start=-40 and the start=[total lines returned from previous call]

also supports /api/logs/nginx, /api/logs/go2rtc

I posted a documentation update just now.

**Comment by Merwenus** [comment]
And what do you do if Frigate uses 100% available CPU and just kills everything inside docker and not even the console come up? And every possible UI is unreachable? How can I get the debug log files in this way?

**Reply by apwebber** [reply]
Yeah. I have a problem where the web interface becomes unreachable. I need to see the logs from the previous session, not just the latest one, which is all the web interface shows. Edit: Seems like the API doesn't retrieve the previous session logs, either.
